server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/RootNode.java

```diff
@@ -41,15 +64,31 @@ public RootNode(Locality locality) {
   }
 
   private List<SyntaxError> updateVariableStructure() {
-    List<Node> node =
+    List<Node> nodes =
         this.getChildren().stream().filter(hasType(NodeType.COPY)).collect(Collectors.toList());
-    node.forEach(it -> removeChild(it));
-    node.forEach(
+    nodes.forEach(it -> removeChild(it));
+    nodes.forEach(
         it ->
             SyntaxTreeUtil.findNodeInRange(this, it.getLocality().getRange().getStart())
                 .orElse(this)
                 .addChild(it));
-
+    for (Node node : nodes) {
+      registerCopyUsage((CopyNode) node);
+    }
     return ImmutableList.of();
   }
+
+  private void registerCopyUsage(CopyNode copyNode) {
+    copyDefinitionNodes.putIfAbsent(
+        copyNode.getCopyBookName(),
+        new CopyDefinitionNode(
+            Locality.builder()
+                .range(new Range())
+                .uri(PREF_IMPLICIT + IMPLICIT_PATH + copyNode.getCopyBookName())
+                .build(),
+            copyNode.getCopyBookName()));
```

Comment:
Won't it change the behavior of calling the "Go to Definition" on SQLCA? Now it should show "No definition found"

Reply:
There is likely a bug from development branch for this particular case. We will open a defect and take it separately. As refactoring is not introducing that defect , I am marking this comment as resolved.